ebml/EbmlElement.h

```diff
@@ -261,16 +263,38 @@ class DllApi x : public BaseClass { \
 #define INVALID_FILEPOS_T 0
 #endif
 
+class EBML_DLL_API EbmlDocVersion {
+  public:
+    using num_version = unsigned char;
+
+    constexpr EbmlDocVersion(num_version min = 0, num_version max = ANY_VERSION)
+      : minver(min)
+      , maxver(max)
+    {}
+
+    /// @brief constant value to indicate the maximum version matches all versions or the minimum version matches no version
+    static const num_version ANY_VERSION = std::numeric_limits<num_version>::max();
+
+    // the minimum DocType version this element is allowed in
+    // ANY_VERSION if the element is never supported
+    const num_version minver;
+    // the maximum DocType version this element is allowed in
+    // ANY_VERSION if the element is supported in all (known) version
+    const num_version maxver;
```

Comment:
I think this would be a great place to use `std::optional` instead of magic values. Just make `minver` & `maxver` an `std::optional<unsigned int>`. Special values are so C++99 😁

Unfortunately `std::optional` is a feature of C++17, and again, so far we only require C++14. Repeating myself here, but I'd be more than fine with bumping the requirements up to C++17, but if I remember correctly this would have the probably undesired consequence that VLC would now also require C++17 instead of C++14. Or do I misremember?

Reply:
Correct, for now VLC only requires a C++14 compiler to build. That means people using just that will not be able to compile libmatroska within VLC, and have Matroska support disabled.

Reply:
The change you propose is probably better. But until we officially make the switch to C++17 we can't use that. So this should not be blocking.

Reply:
I agree. We can always refactor to use `std::optional` later if we decided to require C++17.